microshift.py

```diff
@@ -0,0 +1,223 @@
+def cleanup_microshift(h: host.Host) -> None:
+    files = ["rhocp.toml", "fast-datapath.toml", "kickstart.ks", "*.tar", "*.iso"]
+
+    for file in files:
+        if os.path.exists(f"./{file}"):
+            os.remove(f"./{file}")
+
+    cmd = f"podman rm -f minimal-microshift-server"
+    output = h.run(cmd)
+    logger.info(output.out)
+
+    cleanup_compose_cli(h)
+
+    cleanup_cache_path = "/var/cache/osbuild-worker/osbuild-store/stage/"
+
+    if os.path.exists(cleanup_cache_path):
+        shutil.rmtree(cleanup_cache_path)
+
+
+# cleans composes generated by composer cli
+def cleanup_compose_cli(h: host.Host) -> None:
+    old_compose = h.run("composer-cli compose status").out
+    old_compose_splitlines = old_compose.splitlines()
+
+    for line in old_compose_splitlines[1:]:
+        compose_id = line.split(" ")[0]
+        if 'FAILED' or 'FINISHED' not in line:
+            cmd = f" composer-cli compose cancel {compose_id}"
+            output = h.run(cmd)
+            logger.info(output.out)
+        cmd = f"composer-cli compose delete {compose_id}"
+        output = h.run(cmd)
+        logger.info(output.out)
+
+
+# uses jinja to generate a kickstart file
+def generate_kickstart(rhel_number: str, uname_m: str) -> None:
+    with open('pull_secret.json', 'r') as f_in:
+        file_contents = f_in.read()
+
+    with open('kickstart.ks.j2', 'r') as f:
+        lines = f.read()
+
+    with open('kickstart.ks', 'w') as f_out:
+        template = Template(lines)
+        f_out.write(template.render(rhel_number=rhel_number, uname_m=uname_m, pull_secret=file_contents))
+
+
+# installs dependencies
+def install_packages(package_name: str, h: host.Host) -> None:
+    cmd = f"sudo dnf install {package_name}"
+    output = h.run(cmd)
+    logger.info(output.out)
+
+
+# generates non static toml files
+def generate_toml_file(content: str, file_name: str, h: host.Host) -> None:
+    with open(file_name, 'w') as file:
+        file.write(content)
+
+
+# generates final iso
+def generate_final_iso(file: str, name_of_final_iso: str, h: host.Host) -> None:
+    cmd = f"sudo mkksiso kickstart.ks {file} {name_of_final_iso}"
+    output = h.run(cmd)
+    logger.info(output.out)
+    logger.info("Microshift iso is ready in final.iso")
+
+
+def change_permissions_to_read_for_all(file: str) -> None:
+    uid = os.getuid()
+    gid = os.getgid()
+
+    os.chown(file, uid, gid)
+    os.chmod(file, 444)
+
+
+def vgrename(h: host.Host) -> None:
+    vgname_command = "vgs --noheadings -o vg_name"
+    vgname_output = h.run(vgname_command)
+    vgname = vgname_output.out.split(" ")[0]
+
+    cmd = f"sudo vgrename {vgname} rhel"
+    output = h.run(cmd)
+    logger.info(output.out)
+
+
+# runs various composer-cli commands
+def composer_cli_cmd(subcommand: str, h: host.Host, file_name: Optional[str] = None) -> str:
+    if file_name is None:
+        cmd = f"sudo composer-cli {subcommand}"
+    else:
+        cmd = f"sudo composer-ci {subcommand} {file_name}"
```

Comment:
Typo? composer-ci -> composer-cli

Reply:
Yes. I found out it does retain state when testing. That's why I have added a few more lines to clean state